lib/1.4/dml-builtins.dml

```diff
@@ -1438,39 +1438,37 @@ template bank is (object, shown_desc, _qname) {
     // this carried semantics in DML 1.2; deprecated in 1.4
     param _confidentiality          = undefined;
 
-    session register *_reginfo_cache;
-    session uint64 _reginfo_cache_size = 0;
-
-    shared method _build_register_cache() {
-        if (_reginfo_cache == NULL) {
-            foreach r in (this._each_register) {
-                if (r.offset != unmapped_offset) {
-                    _reginfo_cache_size++;
-                }
+    shared independent startup memoized method _reginfo_cache()
+        -> (register *, uint64) /* cache, cache_size */ {
+        local uint64 size = 0;
+        foreach r in (this._each_register) {
+            if (r.offset != unmapped_offset) {
+                size++;
             }
-            _reginfo_cache = new register[_reginfo_cache_size];
-            local uint64 num = 0;
-            foreach r in (this._each_register) {
-                if (r.offset != unmapped_offset) {
-                    _reginfo_cache[num] = r;
-                    num++;
-                }
+        }
+        local register *cache = new register[size];
+        local uint64 num = 0;
+        foreach r in (this._each_register) {
+            if (r.offset != unmapped_offset) {
+                cache[num] = r;
+                num++;
             }
         }
+        return (cache, size);
     }
 
     // would be great if we could index a sequence
     shared method _get_register(uint32 reg) -> (register) throws default {
-        _build_register_cache();
-        if (reg >= _reginfo_cache_size) {
+        local (register *cache, uint64 cache_size) = _reginfo_cache();
```

Comment:
The word 'cache' is less accurate now that it's eagerly calculated. Maybe `_reginfo_table` or just `_mapped_registers`?

Reply:
Going with table to explicitly connect it to register info.